### PR TITLE
chore: Use jdk 17 for artifact publish

### PR DESCRIPTION
**Pull request by felipecsl**
Welp https://github.com/Eppo-exp/android-sdk/actions/runs/9571411350/job/26388786755

### CODE REVIEW

File: eppo/build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ plugins {
 }
 
 group = "cloud.eppo"
-version = "3.1.0"
+version = "3.2.0-SNAPSHOT"
```
Comment (felipecsl) [inline eppo/build.gradle:8]: bumping to next version since I already ran the release locally

File: eppo/build.gradle [hunk 2]
```diff
@@ -1,11 +1,12 @@
 plugins {
     id 'com.android.library'
     id 'maven-publish'
+    id 'signing'
```
Comment (felipecsl) [inline eppo/build.gradle:4]: was also missing this

**Review by aarsilv — APPROVED** [review]
Approving assuming it's ok to use JDK 17 to publish something that needs to be version 11 compatible. I assume it is, and that Java here is just being used to run publish tooling.

**Comment by felipecsl** [comment]
@aarsilv actually this version of the android gradle plugin requires java 17
```
* What went wrong:
An exception occurred applying plugin request [id: 'com.android.library']
> Failed to apply plugin 'com.android.internal.library'.
   > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
      Your current JDK is located in /opt/hostedtoolcache/Java_Adopt_jdk/11.0.23-9/x64
      You can try some of the following options:
       - changing the IDE settings.
       - changing the JAVA_HOME environment variable.
       - changing `org.gradle.java.home` in `gradle.properties`.
```

**Pull request merged**